### PR TITLE
Add Changesets and GitHub Actions release workflow

### LINKED ISSUE #5755: Use Changesets to handle versioning and releasing
Add Changesets to the monorepo to achieve the following outcomes:

- easier cross-package versioning
- simple CLI to append to package `CHANGELOG.md`
- setup for #5756
- replace `Skip changelog` label with Changesets action to check for changelog entry

### PR DESCRIPTION
**Pull request by sam-b-rose**
### WHY are these changes introduced?

Fixes #5755 

### WHAT is this pull request doing?

Use Changesets to handle versioning and releasing.

**Adds root-level npm scripts**
 - [x] `changeset` for adding a new changelog entry
 - [x] `version-packages` for applying changelog entries to bumping package versions
 - [x] `release` for building packages and releasing to npm 

**Updates GitHub actions**
- [x] Adds `release.yml` GitHub Action workflow for creating a release PR
- [x] Remove ProBot changelog check
- [x] Remove `*.shipit.yml` files

**Updates documentation**
 - [x] Remove `Versioning and changelog.md`
 - [x] Migrate versioning and changelog content to `.github/CONTRIBUTING.md`
 - [x] Update related links in other docs and issue templates
 - [x] Add `changeset` task to `README.md`

### CODE REVIEW

File: documentation/Releasing.md [hunk 1]
```diff
@@ -1,6 +1,16 @@
-# Releasing in the Monorepo
+# Releasing
 
-The process for releasing the monorepo's contents is in active development. This document captures the current process for each piece.
+Polaris uses Changsets to handle releasing the following npm packages:
+
+- [polaris-icons](/polaris-icons)
+- [polaris-react](/polaris-react)
+- [polaris-tokens](/polaris-tokens)
+- [stylelint-polaris](/stylelint-polaris)
```
Comment (alex-page) [inline documentation/Releasing.md]:
```suggestion
Polaris uses Changsets to handle releasing the npm packages in repository.
```

I think focusing this on releasing NPM packages makes it easier to add more NPM packages and not worry about docs.

**Comment by sam-b-rose** [comment]
@alex-page @aaronccasanova I think this is g2g! Hopefully the added docs help explain the process a bit better. Happy to revise, though

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -72,48 +92,65 @@ We’ll review your pull request and either merge it, request changes to it, or
...
 1. Run the TypeScript compiler with `yarn type-check`
 1. [Tophat 🎩](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md) your changes locally in Storybook with `yarn dev`
+1. Create a changeset by running `yarn changeset`. [More info](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#adding-a-changeset).
 1. If you haven’t already, [sign a CLA](https://cla.shopify.com/)
 
-### Contributor License Agreement (CLA)
+### Changelog
 
-Each contributor is required to [sign a CLA](https://cla.shopify.com/). This process is automated as part of your first pull request and is only required once. If any contributor has not signed or does not have an associated GitHub account, the CLA check will fail and the pull request is unable to be merged.
+The changelog is prepared using [Changesets](https://github.com/changesets/changesets). You need to add a changeset by running `yarn changeset` which contains what packages should be bumped, their associated semver bump types and some markdown which will be inserted into changelogs.
```
Comment (alex-page) [inline .github/CONTRIBUTING.md]:
```suggestion
The changelog is created with [Changesets](https://github.com/changesets/changesets).
```

Removing this sentence as it is repeated in the next section.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -72,48 +92,65 @@ We’ll review your pull request and either merge it, request changes to it, or
...
+- Use plain language
 
-Run a command in all workspaces with the [`turbo run` command](https://turborepo.org/docs/reference/command-line-reference#turbo-run-task):
+#### Out of scope for the changelog
 
-- `yarn turbo run <task>` will run the chosen Yarn command in each workspace
+Generally, changes related to these topics can be omitted:
 
-This can be helpful for executing scripts across all packages in the monorepo:
+- Style guide
```
Comment (alex-page) [inline .github/CONTRIBUTING.md]:
```suggestion
- polaris.shopify.com
```
Styleguide is really a meaningless legacy term IMO

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -72,48 +92,65 @@ We’ll review your pull request and either merge it, request changes to it, or
...
+- UI Kit (unless noteworthy)
```
Comment (alex-page) [inline .github/CONTRIBUTING.md]:
```suggestion
- Figma UI Kit
```

We could maybe remove this all together? Who would run changeset for the changes in Figma?

**Review by alex-page — APPROVED** [review]
- [x] Visible education in GitHub actions checks that alerts users that changeset is needed
- [ ] Ship releases for packages before this merges so it collects new changesets
- [ ] Documentation and testing of the release candidate process `*@v9.8.0-rc.1`
- [ ] Opinion or implementation for formatting of `CHANGELOG.md` entries
- [ ] Remove `*.shipit.yml` files

Nothing blocking though these can be follow ups 🎉

**Pull request merged**